spec/app/ics-interchain-queries/README.md

```diff
@@ -0,0 +1,325 @@
+
+`Queried Chain`: The chain whose state is being queried. The Queried Chain gets queried via a relayer utilizing its RPC client which is then submitted back to the Querying Chain.
+
+`Cross-chain Querying Module`: The module that implements the Cross-chain Querying protocol. Only the Querying Chain integrates it.
+
+`Height` and client-related functions are as defined in ICS 2.
+
+`CommitmentPath` and `CommitmentProof` are as defined in ICS 23.
+
+`Identifier`, `get`, `set`, `delete`, `getCurrentHeight`, and module-system related primitives are as defined in ICS 24.
+
+## System Model and Properties
+
+### Assumptions
+
+- **Safe chains:** Both the Querying and Queried chains are safe. This means that, for every chain, the underlying consensus engine satisfies safety (e.g., the chain does not fork) and the execution of the state machine follows the described protocol.
+
+- **Live chains:** Both the Querying and Queried chains MUST be live, i.e., new blocks are eventually added to the chain.
+
+- **Censorship-resistant Querying Chain:**  The Querying Chain cannot selectively omit transactions.
+
+- **Correct relayer:** There is at least one correct relayer between the Querying and Queried chains. This is required for liveness. 
+
+### Desired Properties
+
+#### Permissionless
+
+A Querying Chain can query a chain and implement cross-chain querying without any approval from a third party or chain governance. Note that since there is no prior negotiation between chains, the Querying Chain cannot assume that queried data will be in an expected format.
+
+#### Minimal Queried Chain Work
+
+A Queried Chain has to do no implementation work or add any module to enable cross-chain querying. By utilizing an RPC client on a relayer, this is possible.
+
+#### Modular
+
+Adding cross-chain querying should be as easy as implementing a module in your chain.
+
+#### Control Queried Data
+
+The Querying Chain should have ultimate control over how to handle queried data. Like querying for a certain query form/type.
+
+#### Incentivization
+
+A bounty is paid to incentivize relayers for participating in interchain queries.
+
+## Technical Specification
+
+### General Design 
+
+The Querying Chain MUST implement the Cross-chain Querying module, which allows the Querying Chain to query state at the Queried Chain. 
+
+Cross-chain querying relies on relayers operating between both chains. When a query request is received by the Querying Chain, the Cross-chain Querying module emits a `sendQuery` event. Relayers operating between the Querying and Queried chains must monitor the Querying chain for `sendQuery` events. Eventually, a relayer will retrieve the query request and execute it at the Queried Chain. The relayer then submits (on-chain) the result at the Querying Chain. The result is finally registered at the Querying Chain by the Cross-chain Querying module.
```

Comment:
What do you mean, "execute it at the Queried Chain" - query the full node for proof information?

Reply:
I have clarified that:

...execute it, i.e., fetch the data and generate the corresponding proofs, at the Queried Chain.